src/lazy/any_encoding.rs

```diff
@@ -1211,7 +1412,7 @@ mod tests {
 
             assert!(matches!(
                 reader.next(&allocator)?,
-                LazyRawStreamItem::<AnyEncoding>::EndOfStream
+                LazyRawStreamItem::<AnyEncoding>::EndOfStream(_)
```

Comment:
🗺️ The `EndOfStream` item now returns an `EndPosition` that can provide its range and span. This is not especially useful in and of itself, but it allows the `RawStreamItem` type that contains it to also provide a range and span for every item.